src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs

```diff
@@ -39,93 +40,108 @@ public CommandLineConfigurationProvider(IEnumerable<string> args, IDictionary<st
         public override void Load()
         {
             var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-            string key, value;
+            var argArray = Args as string[] ?? Args.ToArray();
```

Comment:
To avoid `System.Linq` you could use `new List(Args)`.

Reply:
Sure, good idea. I'll push that.

I was also considering a simple helper method to perform the conversion without Linq, but didn't want to make the class too noisy, especially with redundant code.

Reply:
If algorithm is changes from operating on current and next to current and previous (and last item is handled after the loop), we can keep the existing enumerator loop without additional copies.

Reply:
Pushed. Out of curiosity, I gauged the performance impact of using `List<string>` vs just casting to `string[]` and the former was only negligibly slower.

```
List<string>
181.9 μs
172.0 μs
181.2 μs
avg: 178.4 μs

string[]
156.9 μs
178.7 μs
153.6 μs
avg: 163.1 μs
```

Reply:
Oops, sorry @am11, didn't see your comment until after a page reload. Would you prefer I do that instead?

Reply:
Ah, yes I witnessed it as well; code review comments do not autorefresh sometimes for unknown reason. -.-
imo, if the less allocate-y algorithm stayed that way without too much effort and code clutter, it would be appreciated.

Reply:
I'll give it a shot.

Reply:
Code's done and working, just adding some comments before I push.

Reply:
Pushed. I broke out the arg parsing into its own private method to reduce code repetition when evaluating in and out of the while loop. Please let me know your thoughts.

Reply:
Looks great, thanks! :)